pkg/identity/cache/local.go

```diff
@@ -45,11 +45,22 @@ func (l *localIdentityCache) bumpNextNumericIdentity() {
 
 // getNextFreeNumericIdentity returns the next available numeric identity or an error
 // The l.mutex must be held
-func (l *localIdentityCache) getNextFreeNumericIdentity() (identity.NumericIdentity, error) {
+func (l *localIdentityCache) getNextFreeNumericIdentity(idCandidate identity.NumericIdentity) (identity.NumericIdentity, error) {
+	// Try first with the given candidate
+	if idCandidate&identity.LocalIdentityFlag != 0 {
```

Comment:
Nit, we could have a method for `NumericIdentity` which would return true or false if a certain identity was local. We already have a `IsReservedIdentity()` method.

Reply:
Turns out the method already exists: `identity.HasLocalScope()`

Reply:
Fixed.